### PR TITLE
add imput and output arguments in all docstrings

### PR DESCRIPTION
**Pull request by amontoison**
I will add a `#### Optional argument(s)` section when #564 will be closed.
The final goal is to also add `#### Keyword arguments` and `#### Stopping conditions` in all docstrings such that we can do a release 1.0.

### CODE REVIEW

File: src/bicgstab.jl [hunk 1]
```diff
@@ -51,6 +51,16 @@ where `kwargs` are the same keyword arguments as above.
 The callback is called as `callback(solver)` and should return `true` if the main loop should terminate,
 and `false` otherwise.
 
+#### Input arguments
+
+* `A`: a linear operator that models a matrix of dimension n.
```
Comment (dpo) [inline src/bicgstab.jl:56]:
```suggestion
* `A`: a linear operator that models a matrix of dimension n;
```

Comment (dpo) [reply src/bicgstab.jl:56]: I don't see where n is defined.

File: src/bicgstab.jl [hunk 2]
```diff
@@ -51,6 +51,16 @@ where `kwargs` are the same keyword arguments as above.
...
+* `b`: a vector of length n.
+
+#### Output arguments
+
+* `x`: a dense vector of length n.
```
Comment (dpo) [inline src/bicgstab.jl:61]:
```suggestion
* `x`: a dense vector of length n;
```

File: src/bicgstab.jl (hunk 1 above)
Comment (amontoison) [reply src/bicgstab.jl:56]: `n` is not a parameter but I use it to say that the solution is of length `n` in the docstring.
When we have multiple solution, which is the case for `TriCG`, `TriMR` or `GPMR`, it helps to give the length of output vectors.

Comment (dpo) [reply src/bicgstab.jl:56]: I know but n doesn't seem to be defined in the docstring. It should say somewhere that Ax=b is a system of size n.

**Comment by dpo** [comment]
> Is there a reason why you use `####` instead of just `#`?

The font size would be obnoxiously large in the docs.

File: src/cgls.jl [hunk 3]
```diff
@@ -42,7 +42,7 @@ Solve the regularized linear least-squares problem
 
     minimize ‖b - Ax‖₂² + λ‖x‖₂²
 
-using the Conjugate Gradient (CG) method, where λ ≥ 0 is a regularization
+of size n × m using the Conjugate Gradient (CG) method, where λ ≥ 0 is a regularization
```
Comment (dpo) [inline src/cgls.jl:45]: m x n

File: src/gpmr.jl [hunk 4]
```diff
@@ -27,7 +27,7 @@ GPMR solves the unsymmetric partitioned linear system
     [ λI   A ] [ x ] = [ b ]
     [  B  μI ] [ y ]   [ c ],
 
-where λ and μ are real or complex numbers.
+of size (n+m) × (n+m) where λ and μ are real or complex numbers.
```
Comment (dpo) [inline src/gpmr.jl:30]: give the size of A and B instead

**Review by dpo — CHANGES_REQUESTED** [review]

File: src/cgls.jl (hunk 3 above)
Comment (amontoison) [reply src/cgls.jl:45]: Do you want to use `m x n` when `A` has more rows than columns or when A is rectangular?

Comment (dpo) [reply src/cgls.jl:45]: You grab `m, n` as the size of A : https://github.com/JuliaSmoothOptimizers/Krylov.jl/pull/647/files/a2e5796af0bdee8d225347cbd9f984fd554a72c2#diff-695bcdad4090fc22200b17203cb5318c0aa0f12e11a6ecdc048ea445ce5a22f1R96

But yes, a matrix is always m x n :-)

File: src/gpmr.jl (hunk 4 above)
Comment (amontoison) [reply src/gpmr.jl:31]: What do you think of:
```
Given matrices `A` of dimension m × n and `B` of dimension n × m,
GPMR solves the unsymmetric partitioned linear system

    [ λIₘ   A  ] [ x ] = [ b ]
    [  B   μIₙ ] [ y ]   [ c ],

of size (n+m) × (n+m) where λ and μ are real or complex numbers.
```
and
```
Given a matrix `A` of dimension m × n, TriCG / TriMR solves the symmetric linear system

    [ τE    A ] [ x ] = [ b ]
    [  Aᴴ  νF ] [ y ]   [ c ],

of size (n+m) × (n+m) where τ and ν are real numbers, E = M⁻¹ ≻ 0, F = N⁻¹ ≻ 0.
```
?

**Pull request merged**